Basic login page frontend

LieselWong opened the pull request:
### Summary <!-- Required -->

This pull request provides initial styling of the login page detailed in the Figma using MUI's reusable component styling. 

- [x] Got approval for [Figma](https://www.figma.com/design/UtPntFLWZwsNvXkRjTxltJ/Registration-Site?node-id=0-1&p=f&t=AshWnXx02qBy4EKU-0) Design
- [x] Implement Login Page
- [ ] Add reusable text-fields, and buttons

### Notes <!-- Optional -->
Need to make reusable text-fields and global styling, right now it's all individual 
Need to make it resizable 
Add some more detailed styling to further match the Figma

CollinWoo committed:
Initial commit

CollinWoo committed:
add frontend boilerplate

LieselWong committed:
Add Contributer Name

CollinWoo committed:
initialize backend

feverdreme committed:
Update README with 'jeffrey'

feverdreme committed:
Update README with jeffrey role

CollinWoo committed:
Merge pull request #3 from bigredhacks/jeffrey

Update README with 'jeffrey'

LieselWong committed:
contributor name

CollinWoo committed:
Merge branch 'main' into LieselWong

CollinWoo committed:
Merge pull request #2 from bigredhacks/LieselWong

Add Contributor Name

CollinWoo committed:
add api CRUD helpers

CollinWoo committed:
add github PR template

CollinWoo committed:
export firestore database reference

CollinWoo committed:
fix emulator issue

CollinWoo committed:
add helpful scripts to root to run frontend and backend locally

CollinWoo committed:
make firebase config public since it doesn't pose a security risk

CollinWoo committed:
add helper install script

CollinWoo committed:
Add wiki info and footer to README.md

CollinWoo committed:
add MUI to frontend

JamesDos committed:
add name to ReadMe

CollinWoo committed:
fix??

CollinWoo committed:
upgrade MUI and Next.js

CollinWoo committed:
Merge pull request #7 from bigredhacks/JamesTu-update-readme

Add Name to Readme

CollinWoo committed:
remove firebase

CollinWoo committed:
add mongodb config

CollinWoo committed:
remove firebase helpers

CollinWoo committed:
fix .env file handling

LieselWong committed:
Basic login page frontend

CollinWoo commented on package.json:
```diff
@@ -8,5 +8,10 @@
   },
-  "license": "ISC"
+  "license": "ISC",
+  "dependencies": {
```
Don't add dependencies to the root `package.json`. This just exists to define commands that make it easier to install packages/run the frontend and backend locally. Packages for the frontend should be installed in `frontend/package.json`.

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+          variant="contained"
+          color="primary" // ✅ Uses theme color
+          disableElevation
+          sx={{
```
Ideally we don't want inline styling for reusable components like buttons. These styles should be defined using[ MUI themes](https://mui.com/material-ui/customization/theming/).

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+        <Box component="form" sx={{ mt: 2 }}>
+          {/* 📧 Email Input Field */}
+          <TextField
+           fullWidth
```
These two text boxes use basically the same styling. Prime example of where using theming is probably a good idea.

CollinWoo commented on frontend/src/app/layout.tsx:
```diff
@@ -19,10 +20,11 @@ export default function RootLayout({
-        </AppRouterCacheProvider>
+      <body>
+        <ThemeProvider theme={theme}>
+          <CssBaseline />
```
What is `CssBaseline` for here? Also why was `AppRouterCacheProvider` removed? Not too familiar with Next.js configs and would like to know more.

CollinWoo commented on backend/package-lock.json:
```diff
@@ -888,11 +888,12 @@
       }
     },
     "node_modules/braces": {
-      "version": "3.0.2",
```
Upgrading backend dependencies should probably be a different PR since it's not very relevant to this frontend feature.

CollinWoo commented on frontend/public/BRHLogo.png:
The BRH logo should have a transparent background since this picture doesn't seem to match the background. Moreover, if we want to tweak the background color in the future, a transparent background will allow us to make the change without having to push a completely different picture.

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+           margin="normal"
+           required
+           sx={{
+             bgcolor: "#fdf7ed", // ✅ Matches the login box's background color
```
The text field background color shouldn't be the same as the surrounding box. This makes the field less visible to the user, especially if they have vision problems.

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+             bgcolor: "#fdf7ed", // ✅ Matches the login box's background color
+             borderRadius: 2,
+             "& .MuiOutlinedInput-root": { 
+               borderRadius: 3, // ✅ Rounds the input field like the design
```
Some of these comments feel excessive, especially the ones about being "like the design". The main purpose of comments should be to document confusing logic in the code, non-obvious design decisions, or todos, which I feel some of these don't do. Maybe tell gpt to be more sparing with these.

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+            fontSize: "16px", // ✅ Adjusts text size to match design
+            py: 1.5, // ✅ Controls vertical padding
+            fontWeight: "bold",
+            backgroundColor: "#FE1736 !important", // ✅ Forces red background
```
Why is `!important` needed here? You shouldn't have to force override colors for something like this...

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
-          </a>
-        </div>
-      </div>
+    // 🔴 Background Gradient (Red Tone)
```
The emojis are cute but I don't think they're a good idea for documentation comments. I'm not even sure if they render properly in vscode...

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+          />
+          
+          {/* 🔒 Password Input Field */}
+          <TextField
```
The password field shouldn't display the actual characters when someone types in it. It should display dots for each character like a traditional password field.

CollinWoo commented on frontend/src/app/page.tsx:
```diff
@@ -1,113 +1,126 @@
+            py: 1.5, // ✅ Controls vertical padding
+            fontWeight: "bold",
+            backgroundColor: "#FE1736 !important", // ✅ Forces red background
+            "&:hover": { backgroundColor: "#D92030 !important" }, // ✅ Darker red on hover
```
Same here

CollinWoo reviewed (changes requested):
Great work on implementing the design! I think this still needs some work before it can be approved, mainly refactoring the components into the theme. This is partially my b since I think I put adding components to the theme as the last step in the list of tasks when it should have been the first.

LieselWong committed:
upgrade node_modules/braces from 3.0.2 to 3.0.3 due to security vulnerability

LieselWong committed:
remove frontend dependencies

LieselWong committed:
add transparent BRHLogo

LieselWong committed:
update with reusable UI components

LieselWong committed:
update eslintrc file to remove next/babelrc error and fix error on LoginLink page

LieselWong commented:
### Summary

- Added reusable components and removed repeatable inline styling for the login page.
- Added dots to password field for security.
- Updated new photo with transparent background.
- Removed unnecessary comments.
- Removed unnecessary edits to frontend dependencies.
- Updated `node_modules/braces` from version 3.0.2 to 3.0.3 due to a security warning.

This pull request implements Foo.

- [x] Implemented reusable text fields for login page.
- [x] Implemented login page.

### Test Plan

- Ran `npm run dev`, `npm run start`, and `npm run build` without major issues (besides next/babelrc configuration failed to load) and saw the results on localhost.
- Tested on Firefox and Chrome.
- tested different window sizes

### Notes
For future note: 
- The entirety of the <paper> component or box with all the login information and logo may need edits for resizability
